src/app/app.module.ts

```diff
@@ -1,18 +1,46 @@
-import { NgModule } from '@angular/core';
-import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core'
+import { BrowserModule } from '@angular/platform-browser'
+import { MatIconModule } from '@angular/material/icon'
+import { MatInputModule } from '@angular/material/input'
+import { MatCardModule } from '@angular/material/card'
 
-import { AppRoutingModule } from './app-routing.module';
-import { AppComponent } from './app.component';
+import { AppRoutingModule } from './app-routing.module'
+import { AppComponent } from './app.component'
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations'
+
+import { HeaderComponent } from './components/header/header.component'
+import { UserLoginComponent } from './components/user-login/user-login.component'
+import { SearchSectionComponent } from './components/search-section/search-section.component'
+import { VideoCourseListComponent } from './components/video-course-list/video-course-list.component'
+import { VideoCourseComponent } from './components/video-course/video-course.component'
+import { FooterComponent } from './components/footer/footer.component'
+import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component'
+import { LogoComponent } from './components/logo/logo.component'
+import { CoursesPageComponent } from './pages/courses-page/courses-page.component'
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    HeaderComponent,
+    UserLoginComponent,
+    SearchSectionComponent,
+    VideoCourseListComponent,
+    VideoCourseComponent,
+    FooterComponent,
+    BreadcrumbsComponent,
+    LogoComponent,
+    CoursesPageComponent,
   ],
   imports: [
     BrowserModule,
-    AppRoutingModule
+    BrowserAnimationsModule,
+    AppRoutingModule,
+    MatIconModule,
+    MatInputModule,
+    MatCardModule,
+    MatCardModule,
```

Comment:
It is a good practice to keep app.module as clean as possible. The best way to do it is to split all of your components, directives etc into smaller modules.

For example here I propose to create a separate module for Material library, that in a nutshell will look like this
```
const materialModules = [
  MatIconModule,
  MatInputModule,
]
@NgModule({
  imports: [...materialModules],
  exports: [...materialModules],
})
export class MaterialModule {}
```

Reply:
Thank you, very good tip :)